File: frontend/src/services/authService.js
import apiClient from './apiClient.js';

const authService = {
  async register(payload) {
    const { data } = await apiClient.post('/auth/register', payload);
    return data;
  },

  async login(payload) {
    const { data } = await apiClient.post('/auth/login', payload);
    return data;
  },

  async refreshToken(refreshToken) {
    const { data } = await apiClient.post('/auth/refresh-token', { refreshToken });
    return data;
  },

  async verifyEmail(token) {
    const { data } = await apiClient.post('/auth/verify-email', { token });
    return data;
  },

  async forgotPassword(email) {
    const { data } = await apiClient.post('/auth/forgot-password', { email });
    return data;
  },

  async resetPassword(token, newPassword) {
    const { data } = await apiClient.post('/auth/reset-password', { token, newPassword });
    return data;
  },
};

export default authService;

File: backend/src/main/java/com/digitaltwin/platform/service/AuthService.java
package com.digitaltwin.platform.service;

import com.digitaltwin.platform.dto.auth.*;

/**
 * Core authentication use-cases: registration, login, token refresh,
 * email verification, and the forgot/reset password flow.
 */
public interface AuthService {

    void register(RegisterRequest request);

    LoginResponse login(LoginRequest request);

    LoginResponse refreshToken(RefreshTokenRequest request);

    void verifyEmail(EmailVerificationRequest request);

    void forgotPassword(ForgotPasswordRequest request);

    void resetPassword(ResetPasswordRequest request);
}
